### PR TITLE
Add io.github.nemuelw.proton-desktop

### PR DESCRIPTION
**Pull request by nemuelw**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  Unofficial desktop application for [Proton](https://proton.me/) that enables Proton users to access Proton mail, calendar, and drive from the desktop.
- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.nemuelw.proton-desktop.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!-- Copyright 2024 Nemuel Wainaina -->
+<component type="desktop-application">
+  <id>io.github.nemuelw.proton-desktop</id>
+
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-only</project_license>
+
+  <name>Proton Desktop</name>
```
Comment (bbhtt) [inline io.github.nemuelw.proton-desktop.metainfo.xml:10]: You cannot use proton as your application name, that's Proton's brand name

File: io.github.nemuelw.proton-desktop.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    <keyword>Proton Drive</keyword>
+  </keywords>
+
+  <screenshots>
+    <screenshot url="https://github.com/nemuelw/proton-desktop/blob/main/scrshots/loading.png"/>
+    <screenshot url="https://github.com/nemuelw/proton-desktop/blob/main/scrshots/mail.png"/>
+    <screenshot url="https://github.com/nemuelw/proton-desktop/blob/main/scrshots/calendar.png"/>
+    <screenshot url="https://github.com/nemuelw/proton-desktop/blob/main/scrshots/drive.png"/>
+  </screenshots>
+
```
Comment (bbhtt) [inline io.github.nemuelw.proton-desktop.metainfo.xml:56]: where did you see this format?

File: io.github.nemuelw.proton-desktop.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+runtime-version: "24.08"
+sdk: org.freedesktop.Sdk
+command: proton-desktop
+
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=wayland
+  - --socket=x11
+  - --socket=fallback-x11
```
Comment (bbhtt) [inline io.github.nemuelw.proton-desktop.yml:12]: all 3 is wrong, if it works on wayland use wayland+fallback-x11, if not only x11

File: io.github.nemuelw.proton-desktop.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  - name: proton-desktop
+    sources:
+      - type: archive
+        url: https://github.com/nemuelw/proton-desktop/releases/download/v1.0.0/proton-desktop-v1.0.0.tar.gz
+        sha512: 3ecc8bbba183d2fce1d337833d681441ebc650234530e8ccad235c109e015f708486935ce244e9679df5574b8648c9dc1620868ce9d0dbbb72221c3a140b0495
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - cp -r ./_internal /app/bin/_internal
+      - cp ./proton-desktop /app/bin/proton-desktop
```
Comment (bbhtt) [inline io.github.nemuelw.proton-desktop.yml:24]: why not build from source?

File: io.github.nemuelw.proton-desktop.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        path: .
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/share/icons/hicolor/256x256/apps
+      - cp io.github.nemuelw.proton-desktop.png /app/share/icons/hicolor/256x256/apps/io.github.nemuelw.proton-desktop.png
+
+  - name: metainfo.xml
+    sources:
+      - type: dir
+        path: .
```
Comment (bbhtt) [inline io.github.nemuelw.proton-desktop.yml:47]: same as above, all 3 should be merged.

**Comment by bbhtt** [comment]
> - [x] I have read and followed all the [Submission requirements](https://docs.flathub.org/docs/for-app-authors/requirements) and the [Submission guide](https://docs.flathub.org/docs/for-app-authors/submission).

Since you checked this, have you seen the note https://docs.flathub.org/docs/for-app-authors/requirements#required-metadata here?

File: io.github.nemuelw.proton-desktop.metainfo.xml (hunk 2 above)
Comment (bbhtt) [reply io.github.nemuelw.proton-desktop.metainfo.xml:56]: https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#screenshots

Comment (bbhtt) [reply io.github.nemuelw.proton-desktop.metainfo.xml:56]: and the links should be direct links to the image, not links to HTML pages.

https://raw.githubusercontent.com/nemuelw/proton-desktop/954ac083884bacab5aab7c2c535e6bc4b15ee9c6/scrshots/loading.png

File: io.github.nemuelw.proton-desktop.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - cp -r ./_internal /app/bin/_internal
+      - cp ./proton-desktop /app/bin/proton-desktop
+
+  - name: proton-desktop.desktop
+    sources:
+      - type: dir
+        path: .
```
Comment (bbhtt) [inline io.github.nemuelw.proton-desktop.yml:29]: don't use dir sources.

File: io.github.nemuelw.proton-desktop.metainfo.xml (hunk 1 above)
Comment (bbhtt) [reply io.github.nemuelw.proton-desktop.metainfo.xml:10]: https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines#no-trademark-violations

File: io.github.nemuelw.proton-desktop.yml (hunk 4 above)
Comment (nemuelw) [reply io.github.nemuelw.proton-desktop.yml:24]: The app may/will be distributed over other channels besides this one, and so it makes sense to just have a centralized build location from where all distribution channels get the already built version.

Comment (hfiguiere) [reply io.github.nemuelw.proton-desktop.yml:24]: why excluding aarch64 as well?

File: flathub.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: disappointing.

File: io.github.nemuelw.proton-desktop.desktop
Comment (hfiguiere) [inline io.github.nemuelw.proton-desktop.desktop:1]: this file should be managed upstream. Note that a desktop app not having this should be considered a bug.

File: io.github.nemuelw.proton-desktop.png
Comment (hfiguiere) [inline io.github.nemuelw.proton-desktop.png:1]: Icon should be upstream

**Comment by hfiguiere** [comment]
why did you close this? it make reviewing more difficult

**Pull request closed without merging**